impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -218,11 +228,24 @@ private CredentialValidationResult checkDirectBinding(UsernamePasswordCredential
         );
 
         // If this doesn't throw an exception internally, the caller dn exists and the password is correct
-        LdapContext ldapContext = createLdapContext(
+        LdapContext ldapContext = null;
+        try {
+          ldapContext = createLdapContext(
                 ldapIdentityStoreDefinition.url(),
                 callerDn,
                 new String(usernamePasswordCredential.getPassword().getValue())
-        );
+          );
+        } catch (Exception e) {
+          if (e instanceof AuthenticationException) {
+            LOGGER.log(Level.SEVERE, "Error validating credential " + e);
+            return INVALID_RESULT;
+          }
+          // TBD depends on InvalidConfigurationException or RuntimeErrorException, let the exception propagate or return NOT_VALIDATED_RESULT
```

Comment:
I think we need to do a better job of distinguishing these cases (runtime vs. config, what the source of the problem is, etc.).